### PR TITLE
jar upload bug fix

### PR DESCRIPTION
**Pull request by ChengJie1053**
jar upload bug fix

### CODE REVIEW

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/AppBuildPipeServiceImpl.java [hunk 1]
```diff
@@ -234,13 +235,21 @@ public void onStart(PipeSnapshot snapshot) {
...
                     new File(
                         String.format(
                             "%s/%d/%s",
                             Workspace.local().APP_UPLOADS(), app.getTeamId(), app.getJar()));
+                if (!localJar.exists()) {
+                  Resource resource =
+                      resourceService.findByResourceName(app.getTeamId(), app.getJar());
+                  if (resource != null && StringUtils.isNotBlank(resource.getResourcePath())) {
+                    localJar = new File(resource.getResourcePath().split(":")[1]);
+                    uploadJar = appUploads.concat("/").concat(localJar.getName());
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/AppBuildPipeServiceImpl.java:249]: We can add the following methods in the Resource entity to conveniently access the name and path.


``` java

    public void setResourcePath(String resourcePath) {
        if (resourcePath == null) {
            throw new IllegalArgumentException("resource path cannot be null.");
        }
        String[] namePath = resourcePath.split(":");
        if (namePath.length != 2) {
            throw new IllegalArgumentException("resource path invalid, format: $name:$path");
        }
        this.resourcePath = resourcePath;
    }

    public String getFileName() {
        if (this.resourcePath == null) {
            return null;
        }
        return resourcePath.split(":")[0];
    }

    public String getFilePath() {
        if (this.resourcePath == null) {
            return null;
        }
        return resourcePath.split(":")[1];
    }

```

Comment (ChengJie1053) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/AppBuildPipeServiceImpl.java:249]: Ok, thanks for reviewing the code

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Resource.java [hunk 2]
```diff
@@ -82,4 +82,29 @@ public class Resource implements Serializable {
   private transient String sortOrder;
 
   private transient String connector;
+
+  public void setResourcePath(String resourcePath) {
+    if (resourcePath == null) {
+      throw new IllegalArgumentException("resource path cannot be null.");
+    }
+    String[] namePath = resourcePath.split(":");
+    if (namePath.length != 2) {
```
Comment (RocMarshal) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Resource.java:91]: @ChengJie1053 Thx for the contribution~
Could we use Utils.require to replace lines `87`, `91` ?

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ResourceServiceImpl.java [hunk 3]
```diff
@@ -204,13 +205,18 @@ public void deleteResource(Resource resource) {
...
-                findResource.getTeamId(),
-                findResource.getResourceName()));
+    String filePath =
+        String.format(
+            "%s/%d/%s",
+            Workspace.local().APP_UPLOADS(),
+            findResource.getTeamId(),
+            findResource.getResourceName());
+
+    if (!new File(filePath).exists() && StringUtils.isNotBlank(findResource.getFilePath())) {
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ResourceServiceImpl.java:215]: We better use FilesUtils.exists(path...)

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Resource.java (hunk 2 above)
Comment (ChengJie1053) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Resource.java:91]: Ok, thanks for reviewing the code

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ResourceServiceImpl.java (hunk 3 above)
Comment (ChengJie1053) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/impl/ResourceServiceImpl.java:215]: Ok, thanks for reviewing the code

**Review by wolfboys — APPROVED** [review]

**Pull request merged**